bluebottle/initiatives/tests/test_api.py

```diff
@@ -1552,3 +1553,66 @@ def test_submitted_initiatives_dont_show_on_map(self):
 
         data = response.json()
         self.assertTrue(len(data), 5)
+
+    def test_locations(self):
+        InitiativeFactory(
+            status='approved',
+            place=GeolocationFactory(position=Point(23.6851594, 43.0579025))
+        )
+        InitiativeFactory(
+            status='approved',
+            place=GeolocationFactory(position=Point(27.6851594, 34.0579025))
+        )
+
+        response = self.client.get(
+            self.url
+        )
+        data = response.json()
+
+        lat = data[0]['position'][0]
+        long = data[0]['position'][1]
+        lat_2 = data[1]['position'][0]
+        long_2 = data[1]['position'][1]
+        self.assertTrue(lat in data[0]['position'])
+        self.assertTrue(long in data[0]['position'])
+        self.assertTrue(lat_2 in data[1]['position'])
+        self.assertTrue(long_2 in data[1]['position'])
+
+    def test_lat_long_omitted(self):
+        InitiativeFactory(
+            status='approved',
+            place=GeolocationFactory(position=Point(23.6851594, 0))
+        )
+        InitiativeFactory(
+            status='approved',
+            place=GeolocationFactory(position=Point(0, 34.0579025))
+        )
+        InitiativeFactory(
+            status='approved',
+            place=GeolocationFactory(position=Point(23.6851594, 34.0579025))
+        )
+
+        response = self.client.get(
+            self.url
+        )
+        data = response.json()
+
+        for i in range(len(data)):
+            if data[i]['position'][0] == 0.0:
+                del data[i]
+                break
+
+    def test_working_cache(self):
+        InitiativeFactory.create_batch(5, status='approved')
+        response = self.client.get(
+            self.url
+        )
+
+        data = response.json()
+        self.assertTrue(len(data), 5)
+
+        InitiativeFactory.create(status='approved')
+        self.assertTrue(len(data), 7)
```

Comment:
1. Make sure you cache is working for tests 
2. After creating an extra initiative you should `self.client.get` again, otherwise you're just looking at old data
3. Then assert you're getting the cached data so `self.assertTrue(len(data), 7)`
Probably step 1 is the hardest and you'll need further assistance. I would have to look through code or google this too.

Reply:
No need to google it, I know how to make the cache work.